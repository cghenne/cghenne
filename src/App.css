html,
body,
div,
span,
applet,
object,
iframe,
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
a,
abbr,
acronym,
address,
big,
cite,
code,
del,
dfn,
em,
img,
ins,
kbd,
q,
s,
samp,
small,
strike,
strong,
sub,
sup,
tt,
var,
b,
u,
i,
center,
dl,
dt,
dd,
ol,
ul,
li,
fieldset,
form,
label,
legend,
table,
caption,
tbody,
tfoot,
thead,
tr,
th,
td,
article,
aside,
canvas,
details,
embed,
figure,
figcaption,
footer,
header,
hgroup,
menu,
nav,
output,
ruby,
section,
summary,
time,
mark,
audio,
video {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
  display: block;
}

body {
  line-height: 1;
}

ol,
ul {
  list-style: none;
}

blockquote,
q {
  quotes: none;
}

blockquote:before,
blockquote:after,
q:before,
q:after {
  content: '';
  content: none;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}

body {
  background-color: #fdfdfd;
}

.container {
  margin: 20px auto;
  width: 280px;
  padding: 20px;
  height: 100%;
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  grid-template-rows: auto;
  grid-template-areas: 'header' 'hello' 'sayHello' 'skills' 'experience' 'education' 'languages' 'interests' 'references';
}
@media (min-width: 46.25em) {
  .container {
    width: 700px;
    grid-template-columns: repeat(7, 1fr);
    grid-template-areas: 'header header header header header header header' 'hello hello hello divider sayHello sayHello sayHello' 'experience experience experience divider skills skills skills' 'experience experience experience divider education education education' 'experience experience experience divider languages languages languages' 'experience experience experience divider interests interests interests' 'experience experience experience divider references references references' 'experience experience experience divider . . .';
  }
}
@media (min-width: 61.25em) {
  .container {
    width: 940px;
  }
}
@media print {
  .container {
    width: 90%;
    grid-template-columns: repeat(7, 1fr);
    grid-template-areas: 'header header header header header header header' 'hello hello hello divider sayHello sayHello sayHello' 'experience experience experience divider skills skills skills' 'experience experience experience divider education education education' 'experience experience experience divider languages languages languages' 'experience experience experience divider interests interests interests' 'experience experience experience divider references references references' 'experience experience experience divider . . .';
  }
}

.box {
  color: #242021;
  font-size: 16px;
  font-family: 'Roboto', sans-serif;
  line-height: 24px;
  letter-spacing: 1px;
  text-align: justify;
  padding-bottom: 40px;
}
.box h3 {
  color: #58595a;
  font-family: 'Open Sans', sans-serif;
  margin-bottom: 20px;
  font-size: 24px;
  text-transform: uppercase;
  font-weight: 300;
}
