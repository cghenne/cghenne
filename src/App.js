import React, { Component } from 'react';

class App extends Component {
  render() {
    return (
      <div>
        Cathleen Ghenne
      </div>
    );
  }
}

export default App;
