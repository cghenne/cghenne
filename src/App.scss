@import './styles/utils.scss';

body {
  background-color: $color-background;

  @media print {
    background-color: transparent;
  }
}

.container {
  margin: $spacing auto;
  width: 280px;
  padding: $spacing;

  height: 100%;
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  grid-template-rows: auto;
  grid-template-areas:
    'header'
    'hello'
    'sayHello'
    'skills'
    'experience'
    'education'
    'volunteer'
    'languages'
    'interests'
    'references';

  @include mq($from: tablet) {
    width: 700px;
    grid-template-columns: repeat(7, 1fr);
    grid-template-areas:
      'header header header header header header header'
      'hello hello hello divider sayHello sayHello sayHello'
      'experience experience experience divider skills skills skills'
      'experience experience experience divider education education education'
      'experience experience experience divider volunteer volunteer volunteer'
      'experience experience experience divider languages languages languages'
      'experience experience experience divider interests interests interests'
      'experience experience experience divider references references references'
      'experience experience experience divider . . .';
  }

  @include mq($from: desktop) {
    width: 940px;
  }

  @media print {
    padding: 0 $spacing;
    width: 90%;
    height: auto;
    margin: auto;
    grid-template-columns: repeat(7, 1fr);
    grid-template-areas:
      'header header header header header header header'
      'hello hello hello printDivider sayHello sayHello sayHello'
      'experience experience experience printDivider skills skills skills'
      'experience experience experience printDivider education education education'
      'experience experience experience divider volunteer volunteer volunteer'
      'experience experience experience printDivider languages languages languages'
      'experience experience experience printDivider interests interests interests'
      'experience experience experience printDivider references references references'
      'experience experience experience printDivider . . .';
  }
}

.box {
  color: $color-primary;
  font-size: $font-size-regular;
  font-family: $font-family-text;
  line-height: 24px;
  letter-spacing: 1px;
  text-align: justify;
  padding-bottom: $spacing * 2;

  @media print {
    font-size: $font-size-small;
    line-height: 18px;
    letter-spacing: 0px;
    color: $color-primary-print;
    padding-bottom: $spacing * 1.5;

    p {
      line-height: 15px;
    }
  }

  h3 {
    color: $color-secondary;
    font-family: $font-family-title;
    margin-bottom: $spacing;
    font-size: $font-size-large;
    text-transform: uppercase;
    font-weight: $font-weight-light;

    @media print {
      font-size: $font-size-x-regular;
    }
  }
}
