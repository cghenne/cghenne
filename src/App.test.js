import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  const wrapper = shallow(<App />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have container class', () => {
    expect(wrapper.find('.container').exists()).toEqual(true);
  });

  it('should render Header', () => {
    expect(wrapper.find('Header').exists()).toEqual(true);
  });

  it('should render Hello', () => {
    expect(wrapper.find('Hello').exists()).toEqual(true);
  });

  it('should render SayHello with showPrivateDetails to false', () => {
    expect(wrapper.find('SayHello').exists()).toEqual(true);
    expect(wrapper.find('SayHello').prop('showPrivateDetails')).toEqual(false);
  });

  it('should render Experience', () => {
    expect(wrapper.find('Experience').exists()).toEqual(true);
  });

  it('should render Skills', () => {
    expect(wrapper.find('Skills').exists()).toEqual(true);
  });

  it('should render Education', () => {
    expect(wrapper.find('Education').exists()).toEqual(true);
  });

  it('should render Languages', () => {
    expect(wrapper.find('Languages').exists()).toEqual(true);
  });

  it('should render Interests', () => {
    expect(wrapper.find('Interests').exists()).toEqual(true);
  });

  it('should render References', () => {
    expect(wrapper.find('References').exists()).toEqual(true);
  });

  it('should render Divider', () => {
    expect(wrapper.find('Divider').exists()).toEqual(true);
  });

  it('should render PrintDivider', () => {
    expect(wrapper.find('PrintDivider').exists()).toEqual(true);
  });

  it('should render SayHello with showPrivateDetails to true when window.onbeforeprint is called', () => {
    window.onbeforeprint();
    wrapper.update();
    expect(wrapper.find('SayHello').prop('showPrivateDetails')).toEqual(true);
  });

  it('should render SayHello with showPrivateDetails to false when window.onafterprint is called', () => {
    window.onafterprint();
    wrapper.update();
    expect(wrapper.find('SayHello').prop('showPrivateDetails')).toEqual(false);
  });
});
