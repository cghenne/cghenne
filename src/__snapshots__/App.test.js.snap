// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<div
  className="container"
>
  <Header />
  <Hello />
  <SayHello
    showPrivateDetails={false}
  />
  <Experience />
  <Skills />
  <Education />
  <Volunteering />
  <Languages />
  <Interests />
  <References />
  <Divider />
</div>
`;
